internal/httpapi/httpapi.go

```diff
@@ -123,10 +130,50 @@ func (s *Server) Serve(ctx context.Context) {
 		}()
 	}
 
+	c := http.Client{}
+	var ingestReady, statusReady bool
+	for {
+		if !ingestReady && s.cfg.EnableIngest {
+			ingestReady = s.isGetSuccessful(c, fmt.Sprintf("http://%s:%d%s", s.cfg.HostIngest, s.cfg.PortIngest, ingestAPIPathReady))
+		}
+		if !statusReady && s.cfg.EnableStatus {
+			statusReady = s.isGetSuccessful(c, fmt.Sprintf("http://localhost:%d%s", s.cfg.PortStatus, statusAPIPathReady))
+		}
+
+		if s.allReadyOrDisabled(ingestReady, statusReady) {
+			break
+		}
+		time.Sleep(readinessProbeRetryBackoff)
+	}
 	close(s.readyCh)
+
 	<-ctx.Done()
 }
 
+func (s *Server) allReadyOrDisabled(ingestReadyOrDisabled, statusReadyOrDisabled bool) bool {
```

Comment:
I would also change `ingestReadyOrDisabled` to `ingestReady` (same for status) 😅

Reply:
done